scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -276,6 +301,242 @@ private static void unzipReportResources(InputStream source, File target) throws
+                }
+                scanTomlFile.setAnalyzer(analyzer);
+            }
+        });
+
+        Toml rulesTable = scanTomlDocumentContent.getTable(RULES_TABLE).orElse(null);
+        if (rulesTable != null) {
+            TomlValueNode include = rulesTable.get("include").orElse(null);
+            TomlArrayValueNode includeArray;
+            if (include != null && include.kind().equals(TomlType.ARRAY)) {
+                includeArray = (TomlArrayValueNode) include;
+                List<String> rulesToInclude = includeArray.toNativeValue()
+                        .stream()
+                        .map(Object::toString)
+                        .toList();
+
+                rulesToInclude.forEach(ruleToInclude -> {
+                    ScanTomlFile.RuleToFilter ruleToFilter = new ScanTomlFile.RuleToFilter(ruleToInclude);
+                    scanTomlFile.setRuleToInclude(ruleToFilter);
+                });
+            }
+
+            TomlValueNode exclude = rulesTable.get("exclude").orElse(null);
+            TomlArrayValueNode excludeArray = null;
+            if (exclude != null && exclude.kind().equals(TomlType.ARRAY)) {
+                excludeArray = (TomlArrayValueNode) exclude;
+                List<String> rulesToExclude = excludeArray.toNativeValue()
+                        .stream()
+                        .map(Object::toString)
+                        .toList();
+
+                rulesToExclude.forEach(ruleToExclude -> {
+                    ScanTomlFile.RuleToFilter ruleToFilter = new ScanTomlFile.RuleToFilter(ruleToExclude);
+                    scanTomlFile.setRuleToExclude(ruleToFilter);
+                });
+            }
+        }
+        return scanTomlFile;
+    }
+
+    /**
+     * Downloads the remote JAR file to the target directory.
+     *
+     * @param targetDir     the target directory
+     * @param fileName      the remote file name
+     * @param remoteJarFile the remote JAR file path
+     * @param outputStream  the print stream
+     * @return the path of the downloaded remote JAR file
+     */
+    private static String loadRemoteJAR(Path targetDir, String fileName, URL remoteJarFile, PrintStream outputStream) {
+        Path cachedJarPath = targetDir.resolve(fileName + JAR_PREDICATE);
+        if (Files.exists(cachedJarPath)) {
+            outputStream.println("Loading " + fileName + JAR_PREDICATE + " from cache...");
+            return cachedJarPath.toAbsolutePath().toString();
+        }
+        try {
+            FileUtils.copyURLToFile(remoteJarFile, cachedJarPath.toFile());
+            outputStream.println("Downloading remote JAR: " + remoteJarFile);
+            return cachedJarPath.toAbsolutePath().toString();
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to download remote JAR file with exception: " + ex.getMessage());
+        }
+    }
+
     private ScanUtils() {
     }
```

Comment:
Shall we move the private constructor to the top. It will show that the class is a util class to anyone who reads it.